### ISSUE TITLE
twisted.test.test_text fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8666]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8666|
|Type|enhancement|
|Created|2016-07-24 14:40:53Z|
|Branch|https://github.com/twisted/twisted/tree/8666-rodrigc-text|

If I run:

```
python3 bin/trial twisted/test/test_text.py
```

I get:

```
twisted/test/test_text
  py ...                                                                [ERROR]

===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/Users/crodrigues/twisted_win/twisted/trial/runner.py", line 781, in loadByName
    return self.suiteFactory([self.findByName(name, recurse=recurse)])
  File "/Users/crodrigues/twisted_win/twisted/trial/runner.py", line 682, in findByName
    return self.loadFile(_name, recurse=recurse)
  File "/Users/crodrigues/twisted_win/twisted/trial/runner.py", line 652, in loadFile
    module = SourceFileLoader(name, fileName).load_module()
  File "<frozen importlib._bootstrap_external>", line 383, in _check_name_wrapper
    
  File "<frozen importlib._bootstrap_external>", line 805, in load_module
    
  File "<frozen importlib._bootstrap_external>", line 664, in load_module
    
  File "<frozen importlib._bootstrap>", line 268, in _load_module_shim
    
  File "<frozen importlib._bootstrap>", line 693, in _load
    
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
    
  File "<frozen importlib._bootstrap_external>", line 660, in exec_module
    
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "twisted/test/test_text.py", line 8, in <module>
    from cStringIO import StringIO
builtins.ImportError: No module named 'cStringIO'

twisted/test/test_text.py
-------------------------------------------------------------------------------
Ran 1 tests in 0.017s

FAILED (errors=1)

```

<details><summary>Searchable metadata</summary>

```
trac-id__8666 8666
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8666_rodrigc_text 8666-rodrigc-text
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469371253787738 1469371253787738
changetime__1469749172783638 1469749172783638
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/403

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

mark probably should have gotten some credit for a review here :)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @glyph|
|-|-|

Changes look good. Someone please merge :)

Not sure whom to assign this one... so I went for last contributor

It looks like codecov.io is broken again for this PR as in buildbot [1] I can see that the tests are executed, but it codecov [2] the test_text.py is not reported

[1] https://buildbot.twistedmatrix.com/builders/ubuntu16.04-py3.5-coverage/builds/967/steps/default/logs/stdio

[2] https://codecov.io/gh/twisted/twisted/tree/3713a4b5e84a1e22e178d8ac4efb75e4bb11c200/twisted/python/test

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph commented|
|-|-|

Thanks for the review adi, and thanks for taking the time to check the coverage manually.  Hopefully we can work with codecov support to figure this out eventually so we can rely on it; I am really hoping we won't have to invent our own thing there.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub set status to `closed`|
|-|-|

In changeset 4e225c70e9f24af4166fb625d2c0d6645f159c71
```
#!CommitTicketReference repository="" revision="4e225c70e9f24af4166fb625d2c0d6645f159c71"
Merge pull request #403 from twisted/8666-rodrigc-text

Author: glyph
Reviewer: adiroiban, rodrigc
Fixes: #8666

Fix test_text.py on Python 3
```
